mwsimpleedittypes/differ.py

```diff
@@ -145,20 +145,26 @@ def unnest(self, lang='en'):
         and splits it into its component pieces to then identify what has changed between revisions.
         """
         nodes = []
+        # Using string mixin methods such as wt.find(x) for subnodes in the for
+        # loop below means doing str(wt) everytime, which is expensive because
+        # it recursively converts each node to str. Better to create a string using
+        # the node's text and use built-in string methods instead.
+        node_str = self.text
         if self.ntype == 'Gallery':
             # strip leading / trailing gallery tags so parser correctly parses everything in between
             # otherwise links, formatting, etc. is treated as text
             gallery_start = self.text.find('>')
             gallery_end = self.text.rfind('<')
             try:
                 # the <br> is a hack; it'll be skipped in the ifilter loop; otherwise first image skipped
-                wt = mw.parse('<br>' + self.text[gallery_start+1:gallery_end])
+                node_str = '<br>' + self.text[gallery_start+1:gallery_end]
+                wt = mw.parse(node_str)
             except Exception:  # fallback
-                wt = mw.parse(self.text)
+                wt = mw.parse(node_str)
```

Comment:
I'm going to merge this because I have some additional clean-up I'm going to do but just a note that I'll add a `node_str = self.text` line before parsing here because if the try clause fails at `wt = mw.parse(node_str)` then `node_str` has already been updated with the `<br>` text and we'll want to reset it here (or it'll just error again)

Reply:
:facepalm: